src/Api/NavigationUtils.ts

```diff
@@ -136,7 +140,15 @@ export const onResolveConditionsResultAction = (
   restArgs: unknown[],
 ): boolean => {
   const activeLeafRoute = getActiveLeafRoute(state)
-  activeLeafRoute.conditionalNavigation = resolveConditionsResult.conditionalNavigationState
+  if (activeLeafRoute.params == null) {
+    // @ts-expect-error -- NOTE: params are read-only in react-navigation types but we need to override them
```

Comment:
then it's not needed here as well, if we add custom params type in WithConditionalNavigationState